### ISSUE TITLE
[Feature Request] Allow customer to add/configure/remove domains

### ISSUE THREAD

**Issue by dphi** [Froxlor/Froxlor#729]
# Summary
Allow customers in Froxlor to add/configure/remove domains. (Not just subdomains as today.)

Pre-populate the zone configuration with sensible defaults for zone hosting, email, web and any combiation of them. 

Optional: Allow an easy integration with a domain registrar.

Thanks

**Comment by d00p** [comment]
> Allow customers in Froxlor to add/configure/remove domains. (Not just subdomains as today.)

no, it's intended that this is not possible for customers as he can't and shouldn't be able to set specific settings (see differences admin-domain-add and customer-domain-add)

> Pre-populate the zone configuration with sensible defaults for zone hosting, email, web and any combiation of them.

meaning? If DNS is used, a zone file with everything needed is added. So, please be more specific

> Optional: Allow an easy integration with a domain registrar.

very old topic, there are literally thousands of registrars - so, easiest way of integration I could think of was an API....and there you go, you already have that!

**Comment by dphi** [comment]
Thanks for taking the time to review my request.

> > Allow customers in Froxlor to add/configure/remove domains. (Not just subdomains as today.)
> 
> no, it's intended that this is not possible for customers as he can't and shouldn't be able to set specific settings (see differences admin-domain-add and customer-domain-add)
> 
> > Pre-populate the zone configuration with sensible defaults for zone hosting, email, web and any combiation of them.
> 
> meaning? If DNS is used, a zone file with everything needed is added. So, please be more specific

Sure: I meant sensible defaults for all values required to bridge the gap between admin-domain-add and customer-domain-add. - These of course need to be defined by the admin.

> > Optional: Allow an easy integration with a domain registrar.
> very old topic, there are literally thousands of registrars - so, easiest way of integration I could think of was an API....and there you go, you already have that!
I'll review the API - thanks for the hint.

**Comment by d00p** [comment]
> Sure: I meant sensible defaults for all values required to bridge the gap between admin-domain-add and customer-domain-add. - These of course need to be defined by the admin.

This means a lot of changes and additional settings, maybe in later versions

**Comment by MGertz** [comment]
It would be nice to have the option to set how many domains each customer could add.

Let some of the settings only be set by admins. so customer only can set where in their home folder the new domain should point. 

please add this feature into froxlor.

**Comment by d00p** [comment]
Amount of domains that can be added it the least problem. Mostly it's server related settings like, which IP is the domain referring to. This would lead to more settings like assigning usable ip addresses to customers etc. as said earlier, this means a lot of changes...
